src/reducers/selectorLocations.js

```diff
@@ -0,0 +1,26 @@
+import Immutable from 'immutable';
+
+import {SelectorLocations} from '../records';
+
+const emptyMap = new Immutable.Map();
+
+export default function reduceProjects(stateIn, action) {
+  let state;
+
+  if (stateIn === undefined) {
+    state = emptyMap;
+  } else {
+    state = stateIn;
+  }
+
+  switch (action.type) {
+    case 'UPDATE_SELECTOR_LOCATIONS':
+      return state.set(
+        'selectors',
+        new SelectorLocations(action.payload.selectors),
```

Comment:
Although there are some specific exceptions, we usually want to store all non-scalar data in Redux using `Immutable` data structures—so here we should convert each list of locations into an `Immutable` collection (probably a little `Record` representing the location).